internal/indexcoord/index_coord.go

```diff
@@ -752,53 +774,117 @@ func (i *IndexCoord) tsLoop() {
 	}
 }
 
-// recycleUnusedIndexFiles is used to delete useless index files, including lower version index files and index files
-// corresponding to the deleted index.
+func (i *IndexCoord) releaseLockAndReducePriority(ctx context.Context, meta *indexpb.IndexMeta) error {
+	if meta.State != commonpb.IndexState_InProgress && meta.NodeID != 0 {
+		if err := i.tryReleaseSegmentReferLock(ctx, meta.IndexBuildID, meta.NodeID); err != nil {
+			log.Error("release segment reference lock failed", zap.Int64("buildID", meta.IndexBuildID),
+				zap.Int64("nodeID", meta.NodeID), zap.Error(err))
+			return err
+		}
+		i.nodeManager.pq.IncPriority(meta.NodeID, -1)
+		return i.metaTable.ResetNodeID(meta.IndexBuildID)
```

Comment:
first do persistence then IncPriority

Reply:
I think currently put all logic in indexcoord.go is weird.
There should be indexBuild manager to handle all logic.
Meta.go should be another thin layer of persistent storage